docs/how_to/21-headless.rst

```diff
@@ -0,0 +1,114 @@
+
+.. code-block:: python
+
+    urlpatterns = [
+        path('admin/', admin.site.urls),
+        # path('', include('cms.urls'))  # Remove this line
+    ]
+
+Now, django CMS will be fully accessible through the admin interface, but the
+frontend will not be served. Once, you add an API endpoint, this will be the
+only way to access the content.
+
+.. note::
+
+    You can also run a hybrid mode where you serve **both** the HTML pages
+    and the content via an API, say, for an app. In this case, keep the django CMS' URLS and just add the
+    API to your traditional project.
+
+
+To add an API endpoint, you can use the ``djangocms-rest`` package, for example.
+This package provides a REST API for django CMS. To install it, run:
+
+.. code-block:: bash
+
+    pip install djangocms-rest
+
+Then, add the following to your ``urls.py`` file:
+
+.. code-block:: python
+
+    urlpatterns = [
+        path('admin/', admin.site.urls),
+        path('api/', include('djangocms_rest.urls')),
+    ]
+
+
+.. note::
+
+    Django CMS does not force you to use the ``djangocms-rest`` package. You can
+    use any other package that provides an API for django CMS, with
+    a different API such as GraphQL, for example.
+
+    If you are using a different API package, you will need to follow the
+    instructions provided by that package.
+
+
+**************************
+ Headless using templates
+**************************
+
+In traditional Django CMS, placeholders are defined in the templates and they
+represent the regions where your plugins (the content) will be rendered. This
+is easily done via using ``{% placeholder "placeholder_name" %}`` in your
+Django templates.
+
+If you keep the ``CMS_TEMPLATES`` setting in your project, you still will be
+using templates to render the content when editing and previewing in headless
+mode. In this case, the templates will be used to identify the placeholders of
+a page.
+
+This scenario requires templates to be present in the project for the benefit
+of the editors only.
+
+
+****************************
+ Headless without templates
+****************************
+
+However, when running Django CMS headlessly without templates, you fully
+decouple the front-end presentation layer (which includes templates) from the
+CMS, and the configuration of placeholders must be handled differently.
+
+First, set the ``CMS_TEMPLATES`` setting to an empty list in your project's
+``settings.py`` file (or remove it entirely):
+
+.. code-block:: python
+
+    CMS_TEMPLATES = []
+
+Then, you can define the placeholders using the ``CMS_PLACEHOLDERS`` setting:
+
+.. code-block:: python
+
+    CMS_PLACEHOLDERS = (
+        ('single', ('content'), _('Single placeholder')),
+        ('two_column', ('left', 'right'), _('Two columns')),
+    )
+
+The ``CMS_PLACEHOLDERS`` setting is a list of tuples. Each tuple represents a
+placeholder configuration. The first element of the tuple is the name of the
```

Comment:
I suggest we link to `CMS_PLACEHOLDER_CONF` (https://docs.django-cms.org/en/latest/reference/configuration.html#cms-placeholder-conf) here, it wasn't intuitive to me at first glance what "placeholder configuration" means.